frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveImage/index.tsx

```diff
@@ -328,375 +273,115 @@ function renderDots({
   return chart;
 }
 
-const NAME_SPACE_URI = "http://www.w3.org/2000/svg";
```

Comment:
Here and below, moved SVG rendering functions into `/utils.ts`

Net change in this file is adding MutationObserver and wrap `download` function in a debounce